wordpress-mobile/WordPress-Android
# Copying QS data with indexes

This PR copies QuickStart data keeping local indexes so that it's easier to keep the status without the need of complex mappings.

## To test
- Install WordPress login with a user account that has more than one site; also enable the jetpackProviderSyncFeatureConfig from the debug menu
- Start quickstart (both for a new .com site you just created and for an existing site) and go over some of the steps; also leave at least 1 site without onboarding into quickstart at all (no card on the dashboard)
- Set the flag isFeatureFlagEnabled to true on SharedLoginResolver.kt, UserFlagsResolver.kt and BloggingRemindersResolver.kt (this cannot be done using the Debug settings screen because that is only available when the user is already logged in);
- Install Jetpack using the same variant (e.g. if installed WordPress version was jalapenoDebug, install Jetpack jalapenoDebug);
- Jetpack should login automatically after launching, and home screen should be shown;
- Check that the status of the quickstart on all the sites you started it in WordPress is there and correct; also check no quickstart card is on the site where you did not onboarded on quickstart.
- Try to complete the remaining quickstart steps and check they are executed correctly
- Dismiss the quickstart when it's finished; also close and open the app again and check the quickstart is in the correct status or not present any more depending on the state you left it.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
Added more coverage to the existing UserFlagsResolverTest and ResolverUtilityTest unit test

PR submission checklist:
